### PR TITLE
feat: add an empty bb class that other classes can inherit from

### PR DESCRIPTION
**Pull request by LordMidas**
So a typical inheriting usage would look like:
```squirrel
this.my_new_bb_class <- ::inherit(::MSU.BBClass.Empty, {
    m = {
    },
    // my code
});
```

### CODE REVIEW

File: scripts/config/msu/classes/empty_bb_class.nut [hunk 1]
```diff
@@ -0,0 +1,6 @@
+this.empty_bb_class <- {
+	m = {},
+	function create()
+	{
+	}
+}
```
Comment (Enduriel) [inline scripts/config/msu/classes/empty_bb_class.nut:6]: I think this should be in a different folder, maybe bbclasses. I don't think it should be grouped together with Squirrel classes

**Comment by TaroEld** [comment]
Call it dummy_bb_class imp

**Comment by LordMidas** [comment]
@Enduriel , @TaroEld Changes made. Please re-review.

File: scripts/config/!msu.nut [hunk 2]
```diff
@@ -3,5 +3,8 @@
 	ID = "mod_msu",
 	Name = "Modding Standards & Utilities (MSU)",
 	VanillaID = "vanilla",
-	Class = {}
+	Class = {},
+	BBClass = {
+		Dummy = "scripts/config/msu/classes/dummy_bb_class"
```
Comment (Enduriel) [inline scripts/config/!msu.nut:8]: Path is bad now

Comment (TaroEld) [inline scripts/config/!msu.nut:8]: Perhaps just put it in the Class table with DummyBBClass as the name?

Comment (LordMidas) [reply scripts/config/!msu.nut]: No, I think a separate BBClass namespace is much better - as both classes are instantiated in different ways e.g. `local myNewSQClass = ::MSU.Class.WeightedContainer()` vs `local myNewBBClass = ::new(::MSU.BBClass.SomeClass)`.

**Review by Enduriel — APPROVED** [review]
lgtm now

**Comment by LordMidas** [comment]
@Enduriel The only way the inheriting and creating while also allowing hooking the dummy class by using the `::MSU.BBClass.Dummy` can work is if we implement #187.

**Comment by LordMidas** [comment]
Review again please. I've moved it to `scripts/mods/msu/bb_classes` so that it can be both inherited from and hooked via `::MSU.BBClass.Empty`.

**Review by Enduriel — APPROVED** [review]

**Review by TaroEld — APPROVED** [review]

**Comment by LordMidas** [comment]
@Enduriel @TaroEld So, I suggest we remove the `create` function from this "empty" class because apparently that is not required to instantiate any class and several vanilla BB base classes e.g. `skill.nut` do not have a `create` function. If we ever get a bug report that something isn't working because `create` is missing from this empty class, we can then add it.

**Comment by Enduriel** [comment]
Oh wow, thanks for pointing that out, no I'm not okay with the folder structure. The BB_Classes folder is redundant, if it's in scripts and not in !mods_preload or config, it's a bb class. This doesn't need to be stated again.

**Comment by Enduriel** [comment]
I think leaving it in is fine, I don't care enough about it to die on this hill though, I don't think it makes a difference.

**Comment by LordMidas** [comment]
@Enduriel final look please

**Review by Enduriel — APPROVED** [review]

**Pull request merged**